### PR TITLE
[FLINK-26832] Output more status info for JobObserver and BaseObserver

### PR DESCRIPTION
**Pull request by bgeng777**
- Make logger output current and next status in JobObserver/BaseObserver to help developers/users track status from k8s log

### CODE REVIEW

**Comment by bgeng777** [comment]
cc @wangyang0918 @gyfora

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/BaseObserver.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ protected void observeJmDeployment(
         JobManagerDeploymentStatus previousJmStatus =
                 deploymentStatus.getJobManagerDeploymentStatus();
 
-        logger.info("Observing JobManager deployment status");
+        logger.info("Observing JobManager deployment with status: {}", previousJmStatus.name());
```
Comment (gyfora) [inline flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/BaseObserver.java:66]: we could probaby explicitly say `previous` here:
```
Observing JobManager deployment. Previous status: {}
```

Comment (bgeng777) [reply flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/BaseObserver.java]: That is more accurate. Updated.

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/JobObserver.java [hunk 2]
```diff
@@ -81,14 +81,20 @@ private boolean observeFlinkJobStatus(
             flinkAppStatus.getJobStatus().setState(JOB_STATE_UNKNOWN);
             return false;
         }
-
-        updateJobStatus(flinkAppStatus.getJobStatus(), new ArrayList<>(clusterJobStatuses));
-        logger.info("Job status successfully updated");
+        String targetJobStatus =
+                updateJobStatus(flinkAppStatus.getJobStatus(), new ArrayList<>(clusterJobStatuses));
+        logger.info(
+                "Job status successfully updated from {} to {}.",
```
Comment (gyfora) [inline flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/JobObserver.java:87]: To make this really nice we could add a branch:
```
if (changed) {
 "Job status successfully updated from {} to {}."
 } else {
  "Job status ({state}) unchanged."
 }
```

Comment (bgeng777) [reply flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/JobObserver.java]: Updated.

**Pull request merged**